test/langtools/tools/javac/processing/model/type/BasicAnnoTests.java

```diff
@@ -86,12 +89,30 @@ public class BasicAnnoTests extends JavacTestingAbstractProcessor {
     DPrinter dprinter;
     PrintWriter out;
     boolean verbose = true;
+    static Map<String, Class<? extends Annotation>> nameToAnnotation;
+
+    class NameToAnnotationEntry extends  AbstractMap.SimpleEntry<String, Class<? extends Annotation>> {
+        public NameToAnnotationEntry(String key, Class<? extends Annotation> entry) {
+            super(key, entry);
+        }
+    }
 
     @Override
     public void init(ProcessingEnvironment pEnv) {
         super.init(pEnv);
         dprinter = new DPrinter(((JavacProcessingEnvironment) pEnv).getContext());
         out = dprinter.out;
+        // Compile-time mapping to avoid repeated reflective lookups at runtime.
+        nameToAnnotation =
+            Map.ofEntries(new NameToAnnotationEntry("java.lang.Override", Override.class),
+                          new NameToAnnotationEntry("java.lang.annotation.Repeatable", Repeatable.class),
+                          new NameToAnnotationEntry("java.lang.annotation.Target", Target.class),
+                          new NameToAnnotationEntry("BasicAnnoTests.Test", BasicAnnoTests.Test.class),
+                          new NameToAnnotationEntry("BasicAnnoTests.Tests",BasicAnnoTests.Tests.class),
+                          new NameToAnnotationEntry("BasicAnnoTests.TA",   BasicAnnoTests.TA.class),
+                          new NameToAnnotationEntry("BasicAnnoTests.TB",   BasicAnnoTests.TB.class),
+                          new NameToAnnotationEntry("BasicAnnoTests.TC",   BasicAnnoTests.TC.class),
+                          new NameToAnnotationEntry("BasicAnnoTests.TCs",  BasicAnnoTests.TCs.class));
```

Comment:
You could maybe simplify this by obtaining the name from the class parameter.

Reply:
Might be a case a premature optimization, but I wanted to avoid the repeated reflective operations here.

Reply:
Yeah, the map isn't bad in itself, I just have a knee-jerk reaction to mutable static fields ;-)